tests/algorithms/pytorch/test_newton_raphson.py

```diff
@@ -206,6 +210,27 @@ def forward(self, x):
         my_algo.train(datasamples=(x_train, y_train), _skip=True)
 
 
+@pytest.mark.parametrize(
+    "n_samples,y_shape",
+    [(5, 3), (5, 1), (10, 2), (1, 1), (1, 10)],
+)
+def test_prediction_shape(session_dir, perceptron, torch_algo, n_samples, y_shape):
+    x_shape = 15
+
+    x_train = np.zeros([n_samples, x_shape])
+    y_train = np.ones([n_samples, y_shape])
+
+    model = perceptron(linear_n_col=x_shape, linear_n_target=y_shape)
+    my_algo = torch_algo(model=model, batch_size=1)
```

Comment:
What you want to test is the behaviour if batch_size is `None`. Why set it to `1` ?

Reply:
You might want to parametrise it ?